### PR TITLE
[ADD][16.0] stock_picking_restrict_cancel_printed

### PR DESCRIPTION
**Pull request by santostelmo**
Display printed flag on the form view of the transfer.
Raise an error when action cancel is called on a printed stock.picking: `You cannot cancel a picking that is already printed.``

### CODE REVIEW

File: stock_picking_restrict_cancel_printed/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Display printed flag on the form view of the transfer.
+Raise an error when action cancel is called on a printed stock.picking: `You cannot cancel a picking that is already printed.``
```
Comment (sebalix) [inline stock_picking_restrict_cancel_printed/readme/DESCRIPTION.rst]:
```suggestion
Raise an error when canceling a printed stock transfer: `You cannot cancel a transfer that is already printed.`
```

File: stock_picking_restrict_cancel_printed/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_cancel(self):
+        for pick in self:
+            if pick.printed:
+                raise UserError(
+                    _("You cannot cancel a picking that is already printed.")
```
Comment (sebalix) [inline stock_picking_restrict_cancel_printed/models/stock_picking.py]:
```suggestion
                    _("You cannot cancel a transfer that is already printed.")
```

File: stock_picking_restrict_cancel_printed/readme/DESCRIPTION.rst (hunk 1 above)
Comment (santostelmo) [reply stock_picking_restrict_cancel_printed/readme/DESCRIPTION.rst]: Thank you! Description updated

**Review by sebalix — APPROVED** [review]

**Comment by rousseldenis** [comment]
@santostelmo @sebalix Could you look at https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_picking_start ?

As there are similar processes. @jbaudoux

**Comment by jbaudoux** [comment]
> @santostelmo @sebalix Could you look at https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_picking_start ?
> 
> As there are similar processes. @jbaudoux

It's a bit different. Here we want to prevent cancelation if printed (really printed, not started yet)

**Review by cyrilmanuel — APPROVED** [review]
Not an expert But LGTM

File: stock_picking_restrict_cancel_printed/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock picking restrict partner cancel printed",
+    "version": "16.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Inventory",
+    "summary": "Prevent canceling a picking with a printed flag",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
```
Comment (jbaudoux) [inline stock_picking_restrict_cancel_printed/__manifest__.py]:
```suggestion
    "author": "Camptocamp, BCIM, Odoo Community Association (OCA)",
    "maintainers" : ["jbaudoux"],
```

File: stock_picking_restrict_cancel_printed/readme/CONTRIBUTORS.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+* Telmo Santos <telmo.santos@camptocamp.com>
```
Comment (jbaudoux) [inline stock_picking_restrict_cancel_printed/readme/CONTRIBUTORS.rst:1]:
```suggestion
* Telmo Santos <telmo.santos@camptocamp.com>
* Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
```

**Comment by cyrilmanuel** [comment]
hi @rousseldenis, could you merge this PR please ?

**Comment by rousseldenis** [comment]
@santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side

File: stock_picking_restrict_cancel_printed/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Stock picking restrict partner cancel printed",
```
Comment (sebalix) [inline stock_picking_restrict_cancel_printed/__manifest__.py]:
```suggestion
    "name": "Stock Picking - Restrict cancellation if printed",
```

File: stock_picking_restrict_cancel_printed/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "version": "16.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Inventory",
+    "summary": "Prevent canceling a picking with a printed flag",
```
Comment (sebalix) [inline stock_picking_restrict_cancel_printed/__manifest__.py]:
```suggestion
    "summary": "Prevent canceling a stock transfer if printed",
```

**Comment by TDu** [comment]
> @santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side

This commit https://github.com/OCA/stock-logistics-workflow/pull/1568/commits/a34ed2b793018cd5a8d5fbc84e5475888cb397f1 in the backport to v14 adds the option.

**Comment by jbaudoux** [comment]
> > @santostelmo @jbaudoux IMHO this deserves a configuration parameter on picking type side
> 
> This commit [a34ed2b](https://github.com/OCA/stock-logistics-workflow/commit/a34ed2b793018cd5a8d5fbc84e5475888cb397f1) in the backport to v14 adds the option.

@TDu No PR with that commit on this PR? @santostelmo Will you cherry-pick it?

**Comment by TDu** [comment]
Not sure why there is this message. But there is a commit in the backport that adds the option.

**Comment by santostelmo** [comment]
@jbaudoux @TDu I cherry picked the commit with the configuration parameter

**Review by cyrilmanuel — APPROVED** [review]

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**